spec/src/main/asciidoc/rest-endpoints.adoc

```diff
@@ -29,30 +29,46 @@ This section describes the REST-api, that monitoring agents would use to retriev
 A sub-tree that does not contain data must be omitted.
 * A 'shadow tree' that responds to OPTIONS will provide the metadata and tags associated to a metric name.
 
-==== Translation rules for metric names
+==== Translation rules for metric names and handling of tags
 
 The following rules apply only to GET requests:
-* The metric name in JSON is comprised of the metric name appended with the metric tags (if they exist).
-* The metric tags are represented by a comma separated list of key/value tag pairs surrounded by braces `{` and `}`
-
-The following apply to both GET and OPTION requests:
-* Each tag is a key-value-pair in the format of `<key>=<value>`. The list of tags must be sorted alphabetically.
-* If the metric name or tag value contains a special reserved JSON character, these characters must be escaped in the JSON response.
 
+* Tags are appended to the leaf element of the metric's JSON tree with a semicolon `;`.
+* The metric tags are represented by a semicolon `;` separated list of key/value tag pairs.
+* Tagged metrics that contain sub-resources that are registered under the same metric name will share the root element (i.e the metric name).
+* Metrics with tag values containing a semicolon `;` will be converted to underscores `_`
 
-[source]
+For example:
+[source, json]
 ----
 {
-  "cars{colour=red,type=sedan}" : 12,
-  "cars{colour=blue,type=sedan}" : 9,
-  "cars{colour=yellow,type=sedan}" : 16
+ "carsCounter;colour=red" : 0,
+ "carsCounter;colour=blue" : 0,
+ "carsMeter": {
+    "count;colour=red" : 0,
```

Comment:
Should we try to make this consistent, so that backend processors can treat each type the same way?
   carsCounter {
        "colour=red" : 0,
       "colour=blue": 0
   }

Reply:
Hmmm. For a tag-less metric we would have to represent it like 
```
"carsCounter" : {
    "" : 0
}
```
which looks really ugly to me :) and I'm not sure if Collectd would accept this format

Reply:
Ah, as @jmartisk  pointed out that becomes a little bit harder to read.
Also, with that format collectd would build the tagged carsCounter as:
 _carsCounter.colour=red_
 _carsCounter.colour=blue_ 
which doesn't follow the metric;tag=value;tag=value format that graphite/carbon is expecting.